workers/go/activities/activities.go

```diff
@@ -1,7 +1,90 @@
 package activities
 
-import "context"
+import (
+	"context"
+	"go.temporal.io/sdk/activity"
+	"go.temporal.io/sdk/client"
+	"math/rand"
+)
 
-func NoopActivity(ctx context.Context) error {
+type Activities struct {
+	Client client.Client
+}
+
+func (a *Activities) Noop(_ context.Context) error {
 	return nil
 }
+
+type PayloadActivityInput struct {
+	IgnoredInputData []byte
+	// DesiredOutputSize determines the size of the output data in bytes, filled randomly.
+	DesiredOutputSize int
+}
+
+func MakePayloadInput(inSize, outSize int) *PayloadActivityInput {
+	inDat := make([]byte, inSize)
+	rand.Read(inDat)
+	return &PayloadActivityInput{
+		IgnoredInputData:  inDat,
+		DesiredOutputSize: outSize,
+	}
+}
+
+// Payload serves no purpose other than to accept inputs and return outputs of a
+// specific size.
+func (a *Activities) Payload(_ context.Context, in *PayloadActivityInput) ([]byte, error) {
+	output := make([]byte, in.DesiredOutputSize)
+	//goland:noinspection GoDeprecation -- This is fine. We don't need crypto security.
+	rand.Read(output)
+	return output, nil
+}
+
+func (a *Activities) SelfQuery(ctx context.Context, queryType string) error {
+	info := activity.GetInfo(ctx)
+	wid := info.WorkflowExecution.ID
+
+	resp, err := a.Client.QueryWorkflowWithOptions(
+		ctx,
+		&client.QueryWorkflowWithOptionsRequest{
+			WorkflowID: wid,
+			QueryType:  queryType,
+		},
+	)
+
+	if resp != nil && resp.QueryRejected != nil {
+		activity.GetLogger(ctx).Info("failed to query workflow", "queryRejected", *resp.QueryRejected)
+	}
+
+	if err != nil {
+		activity.GetLogger(ctx).Info("failed to query workflow", "Error", err)
+		return err
+	}
+
+	return nil
+}
+
+func (a *Activities) SelfDescribe(ctx context.Context) error {
+	info := activity.GetInfo(ctx)
+	wid := info.WorkflowExecution.ID
+
+	_, err := a.Client.DescribeWorkflowExecution(ctx, wid, "")
+	if err != nil {
+		activity.GetLogger(ctx).Info("failed to describe workflow", "Error", err)
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	return err
```

A bit inconsistent with whether we log errors or not in these activities (I like not doing it personally)

Reply:
Yeah, I meant to get rid of this one thanks